internal/extension/manager.go

```diff
@@ -39,15 +42,15 @@ type Extension interface {
 	Name() string
 }
 
-func NewManager(names []string, location string, logger logr.Logger) (Manager, error) {
+func NewManager(names []string, location string, logger logr.Logger, sync zapcore.WriteSyncer) (Manager, error) {
```

Comment:
I'm thinking I'd make that a `io.Writer` here. That makes it easier to work with, including in tests.